### ISSUE TITLE
Add support to release Linux aarch64 wheels

### ISSUE THREAD

**Issue by odidev** [pyscf/pyscf#1226]
Problem
---------
On aarch64, `pip install pyscf` gives the following error:
```Collecting pyscf
  Using cached pyscf-2.0.1.tar.gz (7.7 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy!=1.16,!=1.17,>=1.13 in /opt/_internal/cpython-3.8.12/lib/python3.8/site-packages (from pyscf) (1.22.0)
Requirement already satisfied: scipy!=1.5.0,!=1.5.1 in /opt/_internal/cpython-3.8.12/lib/python3.8/site-packages (from pyscf) (1.7.3)
Requirement already satisfied: h5py>=2.7 in /opt/_internal/cpython-3.8.12/lib/python3.8/site-packages (from pyscf) (3.6.0)
Building wheels for collected packages: pyscf
  Building wheel for pyscf (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /opt/_internal/cpython-3.8.12/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-sxwoef62/pyscf_99dbddbd7da44484bb71163a64878aba/setup.py'"'"'; __file__='"'"'/tmp/pip-install-sxwoef62/pyscf_99dbddbd7da44484bb71163a64878aba/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-933dx2a1
       cwd: /tmp/pip-install-sxwoef62/pyscf_99dbddbd7da44484bb71163a64878aba/
  Complete output (8 lines):
  running bdist_wheel
  running build
  running build_ext
  Configuring extensions
  cmake -S/tmp/pip-install-sxwoef62/pyscf_99dbddbd7da44484bb71163a64878aba/pyscf/lib -Bbuild/temp.linux-aarch64-3.8
  CMake Error: The source directory "/tmp/pip-install-sxwoef62/pyscf_99dbddbd7da44484bb71163a64878aba/pyscf/lib" does not appear to contain CMakeLists.txt.
  Specify --help for usage, or press the help button on the CMake GUI.
  error: command 'cmake' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pyscf
```

Resolution
-----------
On aarch64, `pip install pyscf` should download the wheels from PyPI.

@sunqm, Please let me know your interest in releasing aarch64 wheels. I can help with this.

**Comment by sunqm** [comment]
For sure I'm interested in adding support to aarch64.

For the error you posted, it's not clear for me based on the output of pip install. If you manually run cmake inside the pyscf/lib directory,
```
cmake -B build pyscf/lib && cmake --build build
```
does it produce any error?

**Comment by susilehtola** [comment]
FWIW I've had no problems in building PySCF on aarch64, which is one of the official Fedora Linux platforms, see e.g. [here](https://kojipkgs.fedoraproject.org//packages/python-pyscf/2.0.1/4.fc36/data/logs/aarch64/build.log) for the last build log

**Comment by odidev** [comment]
> ```
> cmake -B build pyscf/lib && cmake --build build
> ```
> 
> does it produce any error?

Running the above command does not produce any error and builds pyscf successfully. But it takes a lot of time to build from source on AArch64. It would be beneficial for the AArch64 users if the wheels for pyscf are present on PyPI. 

I have raised a [PR ](https://github.com/pyscf/pyscf/pull/1227) for adding aarch64 wheel build support. Could you please have a look at it?

**Comment by sunqm** [comment]
Available in 2.1 release
